controllers/cloud.redhat.com/providers/kafka/strimzi.go

```diff
@@ -188,37 +190,44 @@ func (s *strimziProvider) Provide(app *crd.ClowdApp, c *config.AppConfig) error
 			f, ok := conversionMap[key]
 			if ok {
 				out, _ := f(valList)
-				newConfig[key] = out
+				k.Spec.Config[key] = out
 			} else {
 				return errors.New(fmt.Sprintf("no conversion type for %s", key))
 			}
 		}
+
 		if len(replicaValList) > 0 {
 			maxReplicas, err := utils.IntMax(replicaValList)
 			if err != nil {
 				return errors.New(fmt.Sprintf("could not compute max for %v", replicaValList))
 			}
-			maxReplicaString, err := strconv.Atoi(maxReplicas)
+			maxReplicasInt, err := safecast.Atoi32(maxReplicas)
 			if err != nil {
-				return errors.New(fmt.Sprintf("could not convert to string %v", maxReplicas))
+				return errors.New(fmt.Sprintf("could not convert string to int32 for %v", maxReplicas))
+			}
+			k.Spec.Replicas = maxReplicasInt
+			if k.Spec.Replicas < int32(1) {
```

Comment:
If someone sets replicas to `-1`, this would set it to `3`?  If that's the case, I might prefer to return an error instead and only default to `3` when `Replicas` is `0`.

Reply:
How about using kubebuilder minimum/maximum annotations on our Type definition to prevent them from setting "-1" on a ClowdApp's config?

Reply:
Yes, 💯

Reply:
Done